### DISCUSSION TITLE
Prisma Client as standalone package

### DISCUSSION THREAD

**Discussion by kleinpetr** [prisma/prisma#19114] [Q&A]
Hi,

I am trying to solve some kind of SRP within Prisma intergation.
The main goal is to seaprate prisma client from frontend projects. I've already generate and deployed a standalone package, but then I am facing an issue with `openssl` on different platforms. So I've included lot of them into the schema itself, but with this approach, the package has about 30MB, which is crazy :thinking: 

Is there any other approach how to keep prisma client standalone?

Thanks

**Comment by nurul3101** [comment]
Hello @kleinpetr 👋 

What are you including in your package?
Are you including only the generated Prisma types in the package? Or are you including the Prisma Schema as well?
This repository exports a package, this might be helpful: https://github.com/MarcMogdanz/typed-prisma-package

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my response
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by kleinpetr** [reply]
Hello thanks for reply, currently I've set output in the schema
```
generator client {
..
output          = "../dist"
..
```
and within the package.json I am wrapping just this dist folder
```
"main": "./dist/index.js",
  "files": [
    "dist"
  ],
```

But when I look into your approach, it seems very similar but the only difference (probably) is that you build the prisma client in the Github CI, ok, maybe then you don't have a trouble with openssl if it's using ubuntu for generating Prisma client and later on you build also your app with Github CI with same environment. But what would happened when someone pull your project (the app) which includes this Prisma package on Windows?

Then as I mentioned you should go and update binary options by `windows` and this is what I am talking about, because if you have multiple developers, with multiple OS, the generated prisma client can have even 40-50 MB..

**Reply by nurul3101** [reply]
> Then as I mentioned you should go and update binary options by windows and this is what I am talking about, because if you have multiple developers, with multiple OS, the generated prisma client can have even 40-50 MB.

Can you use Docker? With this approach, all developers would use a Docker container for development. The Prisma client would be built inside the Docker container, ensuring that it's always compatible with the container's OS. This approach requires that all developers have Docker installed and are familiar with it, but it can simplify the development process and ensure consistency across different environments.

**Reply by nurul3101** [reply]
I'll close this GitHub Discussion, please feel free to reopen if needed.
